### PR TITLE
Update iwslt18 recipe

### PR DESCRIPTION
**Pull request by hirofumi0810**
- [x] Move reclist and test-db.yaml to Google Drive
- [x] Update config for the recent update
- [x] Enhance local  
- [x] Context residual
- [x] Add results

### CODE REVIEW

**Comment by ShigekiKarita** [comment]
Let me know when you finish. I will exec `run.sh` in my pc

**Comment by sw005320** [comment]
@hirofumi0810, will you work on context residual or do you want us to review this PR right now?

**Comment by hirofumi0810** [comment]
@sw005320 Please start to review. I'll add results until tomorrow (now decoding).

**Comment by sw005320** [comment]
It seems that the speed perturbation option makes the recipe look complicated. How about using it as default?

File: espnet/bin/asr_train.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def main(cmd_args):
                         help='Number of decoder layers')
     parser.add_argument('--dunits', default=320, type=int,
                         help='Number of decoder hidden units')
-    parser.add_argument('--mtlalpha', default=0.5, type=float,
```
Comment (sw005320) [inline espnet/bin/asr_train.py]: Could you keep the original default value?
This may cause confusions.

File: espnet/bin/asr_train.py [hunk 2]
```diff
@@ -137,13 +137,13 @@ def main(cmd_args):
                         to automatically find maximum hypothesis lengths""")
     parser.add_argument('--minlenratio', default=0.0, type=float,
                         help='Input length ratio to obtain min output length')
-    parser.add_argument('--ctc-weight', default=0.3, type=float,
```
Comment (sw005320) [inline espnet/bin/asr_train.py]: same comment

File: espnet/bin/asr_train.py [hunk 3]
```diff
@@ -137,13 +137,13 @@ def main(cmd_args):
...
+    parser.add_argument('--ctc-weight', default=0.0, type=float,
                         help='CTC weight in joint decoding')
     parser.add_argument('--rnnlm', type=str, default=None,
                         help='RNNLM model file to read')
     parser.add_argument('--rnnlm-conf', type=str, default=None,
                         help='RNNLM model config file to read')
-    parser.add_argument('--lm-weight', default=0.1, type=float,
```
Comment (sw005320) [inline espnet/bin/asr_train.py]: same comment

**Comment by sw005320** [comment]
I'm testing your code for the data preparation part again as there are a lot of changes.

**Comment by sw005320** [comment]
You may need to add the installation of yaml in setup.py
```
# Running on a14
# Started at Wed Jun 12 11:21:33 EDT 2019
# apply-cmvn --norm-vars=true data/train_nodevtest_sp.de/cmvn.ark scp:exp/dump_feats/train_nodevtest_sp.de/feats.1.scp ark:- | copy-feats.py --verbose 0 --out-filetype mat --compress=true --compression-method=2 --write-num-frames=ark,t:/export/a08/shinji/201707e2e/espnet_iwslt18/egs/iwslt18/st1/dump/train_nodevtest_sp.de/deltafalse/utt2num_frames.1 ark:- ark,scp:/export/a08/shinji/201707e2e/espnet_iwslt18/egs/iwslt18/st1/dump/train_nodevtest_sp.de/deltafalse/feats.1.ark,/export/a08/shinji/201707e2e/espnet_iwslt18/egs/iwslt18/st1/dump/train_nodevtest_sp.de/deltafalse/feats.1.scp
apply-cmvn --norm-vars=true data/train_nodevtest_sp.de/cmvn.ark scp:exp/dump_feats/train_nodevtest_sp.de/feats.1.scp ark:-
Traceback (most recent call last):
  File "/export/a08/shinji/201707e2e/espnet_iwslt18/egs/iwslt18/st1/../../../utils/copy-feats.py", line 6, in <module>
    from espnet.transform.transformation import Transformation
  File "/export/a08/shinji/201707e2e/espnet_iwslt18/espnet/transform/transformation.py", line 7, in <module>
    import yaml
ModuleNotFoundError: No module named 'yaml'
# Accounting: time=2 threads=1
# Finished at Wed Jun 12 11:21:35 EDT 2019 with status 1
exp/dump_feats/train_nodevtest_sp.de/dump_feature.1.log (END)
```

**Comment by hirofumi0810** [comment]
@sw005320 I think PyYAML is already included in setup.py.

**Comment by sw005320** [comment]
oh, I see.
I'll reinstall it

**Comment by sw005320** [comment]
The data preparation part of both asr1 and st1 recipes was finished without errors :)

**Comment by hirofumi0810** [comment]
@sw005320 Can you accept this PR after adding results?

**Comment by sw005320** [comment]
Yes. I just found the following errors:
```
asr_train.py: error: unrecognized arguments: --asr-model --mt-model
```

**Comment by hirofumi0810** [comment]
Oh, I'm sorry. That lines should be added in another PR.

**Comment by sw005320** [comment]
@hirofumi0810, could you follow up #841.
Also, I found the following errors during the speech translation inference.
Could you fix this?

```
# Running on c06
# Started at Fri Jun 14 12:19:44 EDT 2019
# asr_recog.py --config conf/decode.yaml --ngpu 0 --backend pytorch --batchsize 0 --recog-json dump/tst2014.de/deltafalse/split16utt/data.1.json --result-label exp/train_nodevtest_sp.de_lc_pytorch_train/decode_tst2014.de_decode/data.1.json --model exp/tra
in_nodevtest_sp.de_lc_pytorch_train/results/model.acc.best
2019-06-14 12:19:45,983 (asr_recog:127) INFO: python path = (None)
2019-06-14 12:19:45,983 (asr_recog:132) INFO: set random seed = 1
2019-06-14 12:19:45,983 (asr_recog:140) INFO: backend = pytorch
/home/shinji/work/201707e2e/espnet_iwslt18/tools/venv/lib/python3.7/site-packages/chainer/_environment_check.py:73: UserWarning:
--------------------------------------------------------------------------------
CuPy (cupy) version 5.0.0 may not be compatible with this version of Chainer.
Please consider installing the supported version by running:
  $ pip install 'cupy>=6.0.0,<7.0.0'

See the following page for more details:
  https://docs-cupy.chainer.org/en/latest/install.html
--------------------------------------------------------------------------------

  requirement=requirement, help=help))
/home/shinji/work/201707e2e/espnet_iwslt18/tools/venv/lib/python3.7/site-packages/sklearn/externals/joblib/externals/cloudpickle/cloudpickle.py:47: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
2019-06-14 12:20:16,032 (deterministic_utils:24) INFO: torch type check is disabled
2019-06-14 12:20:16,042 (asr_utils:288) INFO: reading a config file from exp/train_nodevtest_sp.de_lc_pytorch_train/results/model.json
2019-06-14 12:20:16,043 (asr:475) INFO: reading model parameters from exp/train_nodevtest_sp.de_lc_pytorch_train/results/model.acc.best
2019-06-14 12:20:16,043 (e2e_asr:85) WARNING: Subsampling is not performed for vgg*. It is performed in max pooling layers at CNN.
2019-06-14 12:20:16,043 (e2e_asr:86) INFO: subsample: 1 1 1 1 1 1
2019-06-14 12:20:17,660 (encoders:231) INFO: Use CNN-VGG + BLSTM for encoder
2019-06-14 12:21:00,947 (asr:538) INFO: (1/88) decoding tst2014.en.talkid1443_0000000_0013000
2019-06-14 12:21:01,199 (encoders:168) INFO: VGG2L input lengths: [1298]
2019-06-14 12:21:02,229 (encoders:117) INFO: RNN input lengths: [325]
2019-06-14 12:21:50,999 (decoders:234) INFO: input lengths: 325
2019-06-14 12:21:51,000 (decoders:260) INFO: max output length: 520
2019-06-14 12:21:51,000 (decoders:261) INFO: min output length: 0
2019-06-14 12:26:18,193 (decoders:351) INFO: adding <eos> in the last position in the loop
2019-06-14 12:26:18,212 (decoders:380) INFO: no hypothesis. Finish decoding.
2019-06-14 12:26:18,230 (decoders:400) INFO: total log probability: tensor(0.5660)
2019-06-14 12:26:18,231 (decoders:401) INFO: normalized log probability: tensor(0.0377)
Traceback (most recent call last):
  File "/export/a08/shinji/201707e2e/espnet_iwslt18/egs/iwslt18/st1/../../../espnet/bin/asr_recog.py", line 159, in <module>
    main(sys.argv[1:])
  File "/export/a08/shinji/201707e2e/espnet_iwslt18/egs/iwslt18/st1/../../../espnet/bin/asr_recog.py", line 147, in main
    recog(args)
  File "/export/a08/shinji/201707e2e/espnet_iwslt18/espnet/asr/pytorch_backend/asr.py", line 572, in recog
    new_js[name] = add_results_to_json(js[name], nbest_hyps, train_args.char_list)
  File "/export/a08/shinji/201707e2e/espnet_iwslt18/espnet/asr/asr_utils.py", line 431, in add_results_to_json
    out_dic = dict(js['output'][0].items())
IndexError: list index out of range
# Accounting: time=395 threads=1
# Finished at Fri Jun 14 12:26:19 EDT 2019 with status 1
```

**Comment by sw005320** [comment]
Same errors happened in the ASR recipe. I don't know why the ASR recipe does not work.

**Comment by sw005320** [comment]
I think this is the error that I reported before. Some of them do not have references and the evaluation is failed due to that. I think we have two options
1. remove these data set from the evaluation
2. add "non-evaluation" mode during the inference

I like option 2, but an easier one is option 1.

**Comment by hirofumi0810** [comment]
> I think this is the error that I reported before. Some of them do not have references and the evaluation is failed due to that. I think we have two options
> 
> 1. remove these data set from the evaluation
> 2. add "non-evaluation" mode during the inference
> 
> I like option 2, but an easier one is option 1.

@sw005320 
I chose option 2. I think everythink is OK now.

**Comment by sw005320** [comment]
Thanks! I also confirmed that it's working from your fork. The main training/inference scripts are not finished yet, but I'm merging this PR to avoid to make the PR very large. Anyway, great job, @hirofumi0810!

**Pull request merged**